compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -950,7 +950,6 @@ class Objects(using Context @constructorOnly):
 
         val instance = OfClass(klass, outerWidened, ctor, args.map(_.value), envWidened)
         callConstructor(instance, ctor, args)
```

Comment:
I'd add a comment above `callConstructor` and refer to the test. Something like the following:

```
// The actual instance might be cached without running the constructor. 
// See tests/init-global/pos/cache-constructor.scala
```